### PR TITLE
fix stable diffusion decoder non standard shape issue

### PR DESCRIPTION
**Pull request by shivadbhavsar**
Addressing issue described in #1579 
In simplify algebra, there are certain edge cases where we manipulate instructions in a graph where the resulting shape becomes non-standard (mostly by transpose ops). Example shown in linked issue.

Updates:
1. Always add contiguous in reshape parser to be safe. Unnecessary contiguous instructions are removed early on by the simply_reshapes pass when it looks for nop reshapes. Remaining ones are taken care of by eliminate_contiguous.
2. It's possible for strides corresponding to dim size 1 to be an arbitrary number for standard shapes (since it is still contiguous). These are the instructions that cause problems in simplify_algebra and we do not want nop_reshapes pass to remove contiguous in this case. So even if the input is technically standard, this update forces contiguous op to recompute the strides.

### CODE REVIEW

File: src/include/migraphx/op/contiguous.hpp [hunk 1]
```diff
@@ -48,7 +48,7 @@ struct contiguous
     {
         check_shapes{inputs, *this, true}.has(1);
         auto s0 = inputs.front();
-        if(s0.dynamic() or s0.standard())
+        if(s0.dynamic() or s0.scalar())
```
Comment (pfultz2) [inline src/include/migraphx/op/contiguous.hpp:51]: Contiguous after scalar should be a standard shape nor a scalar shape.

Comment (shivadbhavsar) [reply src/include/migraphx/op/contiguous.hpp]: ok reverted. However this test case in op_shape_test.cpp suggests that contiguous after scalar is scalar:

TEST_CASE(contiguous_shape_scalar)
{
    migraphx::shape output{migraphx::shape::float_type};
    migraphx::shape input{migraphx::shape::float_type};
    expect_shape(output, migraphx::make_op("contiguous"), input);
}

File: test/op_shape_test.cpp [hunk 2]
```diff
@@ -431,7 +431,7 @@ TEST_CASE(contiguous_dyn_shape)
 
 TEST_CASE(contiguous_shape_scalar)
 {
-    migraphx::shape output{migraphx::shape::float_type};
+    migraphx::shape output{migraphx::shape::float_type, {1}};
```
Comment (umangyadav) [inline test/op_shape_test.cpp:434]: Can we add a test inside `shape_test` for standard() where one of the dims is 1 and stride for for that dim is a random value? It should assert that it is a standard shape.

Comment (umangyadav) [reply test/op_shape_test.cpp:434]: Also test add test for `contiguous` in `op_shape_test` where it converts such standard shape to "standard"-standard shape.

Comment (shivadbhavsar) [reply test/op_shape_test.cpp:434]: it seems like contiguous does not convert modify the strides for such a shape. Is that the expected behaviour?

**Review by umangyadav — APPROVED** [review]

**Comment by causten** [comment]
@pfultz2  anything remaining?  I'd like to merge this one soon

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**